test/unit/strainer_factory_unit_test.rb

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class StrainerFactoryUnitTest < Minitest::Test
+  include Liquid
+
+  module AccessScopeFilters
+    def public_filter
+      "public"
+    end
+
+    def private_filter
+      "private"
+    end
+    private :private_filter
+  end
+
+  StrainerFactory.add_global_filter(AccessScopeFilters)
+
+  module LateAddedFilter
+    def late_added_filter(_input)
+      "filtered"
+    end
+  end
+
+  def test_strainer
+    strainer = StrainerFactory.create(nil)
+    assert_equal(5, strainer.invoke('size', 'input'))
+    assert_equal("public", strainer.invoke("public_filter"))
+  end
+
+  def test_stainer_raises_argument_error
+    strainer = StrainerFactory.create(nil)
+    assert_raises(Liquid::ArgumentError) do
+      strainer.invoke("public_filter", 1)
+    end
+  end
+
+  def test_stainer_argument_error_contains_backtrace
+    strainer = StrainerFactory.create(nil)
+    begin
+      strainer.invoke("public_filter", 1)
+    rescue Liquid::ArgumentError => e
```

Comment:
This might not be your code but this test will pass if `strainer.invoke("public_filter", 1)` doesn't raise.

Might I suggest

```ruby
exception = assert_raises(Liquid::ArgumentError) do
  strainer.invoke("public_filter", 1)
end

assert_match(
  /\ALiquid error: wrong number of arguments \((1 for 0|given 1, expected 0)\)\z/,
  exception.message
)

# ...
```